### ISSUE TITLE
Change code quality rating to F, instead of A (widget is probably fake)

### ISSUE THREAD

**Issue by telugu-boy** [kubinka0505/iFunny-Captions#11]
Should be self explanatory. View any file in repo for reasoning

**Comment by telugu-boy** [comment]
Also the maintainability widget

**Comment by kubinka0505** [comment]
1. I will not improve the readability of the code for two reasons.
     1. I don't want its size to exceed - for the moment - over 100 KB
        Therefore, I **will not simplify each existing list comprehension** and split each comment content into separate file - and thus increase its size.
     2. Furthermore, I also don't want to slow down the program by such operation - it will have to load more and more files and - as it is commonly known - [Python is a **very** slow language](https://www.geeksforgeeks.org/what-makes-python-a-slow-language/).
2. > Should be self-explanatory. View any file in repo for reasoning


     The code analyzers I have used **do not analyze code primarily for readability, but for the security**.
     The program repeatedly uses the `exec` command, which, according to the above-mentioned analyzers, is the most heavily intercepted and - together with the `eval` command - [is considered to be the most dangerous by them](https://docs.openstack.org/bandit/1.4.0/plugins/exec_used.html).
     In fact, most uses of this command are found in the [`__init__.pyw`](https://github.com/kubinka0505/iFunny-Captions/blob/master/__init__.pyw#L61=) file and are used to load the rest of the files necessary to use the program.

     If you want to see the rest of the according to me reasonable and real stylistic errors in the program, you can see them [here](https://app.codacy.com/gh/kubinka0505/iFunny-Captions/issues).

     Per your complaint, in the next release I will create a special configuration file that will replace `Config["..."]["..."]` with user-friendly names in all possible other places.
5. > Also the maintainability widget

    *Maintainability* is a subjective factor that determines **ease to maintain, not modification of the project**.
By logic - the bigger the project is, the more variables, operations and other factors it requires to sustain its proper development.

I understand your frustration, but the lack of motivation and the level of sophistication of the project, and my personal programming experience effectively reinforces my belief that **code readability is far less important than code performance** - in the user community, programmers make up only a small percentage.

**Comment by telugu-boy** [comment]
Baseless and nonsensical lmao. Code size has zero to do with anything in Python, not even its speed. 100kb is an arbitrary number. Express what you're writing in Python **properly** and **clearly** instead of lying to yourself that obfuscating your source code is somehow good - performance will follow. Python isn't as slow as you think.

Code maintainability is literally defined around how easy your code is to modify, and your code isn't exactly a candidate for proper static security analysis due to your strange practices and dynamic loading which analysis tools probably don't understand. _By logic_ your definition of code complexity supports this in the fact that people need to _understand_ and _modify_ your code.

The "actual problems" in your code as determined by Codacy are extremely primitive best practices, and shows its inability to understand what your code does.

**Shitty, unreadable and insecure** (sure, exec is "intercepted", but is the literal code not? Makes zero sense lol. Literally all you're doing is confusing static analysis tools, and everyone around you) code has **NO bearing on its performance**, and you're misguided for thinking you're optimizing anything while making your code impossible to test or maintain or debug by anybody else except you. Do you have any benchmarks of your code versus any _sane_ code (you know, code that loads modules with `import` and not `exec` 🤣)?

TLDR:

- Realize that your code is not good, and that reflects your experience. Not what you say your experience is.
- Take the initiative to **learn Python properly** to refactor this kind-of-functioning abomination of code, into something which won't automatically classify you as insane or a troll when read by any normal Python programmer.
- Realize that "code readability is far less important than code performance" is bullshit 99% of the time when you're using _Python_, which you claim is slow, and without any benchmarks.
- "in the user community, programmers make up only a small percentage". Then don't make your code open source if you don't care about contribution from others or their tinkering. You're misleading anybody who wants to learn Python, and yourself.
- Stop thinking (**extremely**) shitty code is fast code. It's not. 🤡

I'd be surprised if any programmer that has read PEP8 doesn't voluntarily forsake programming and vow to never touch a keyboard after reading your code.
